cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/coprocessor/ConsumerConfigCache.java

```diff
@@ -232,12 +235,28 @@ public void run() {
     refreshThread.start();
   }
 
-  public static ConsumerConfigCache getInstance(Configuration hConf, byte[] tableName,
+  /**
+   * Sets an attribute to the given {@link Scan} object. Instead of calling {@link Scan#setAttribute(String, byte[])}
+   * directly, it uses reflection to call the method. This is because the return type for the setAttribute method
+   * is different in different HBase version.
+   */
+  private void setScanAttribute(Scan scan, String name, byte[] value) {
```

Comment:
This does not currently have access to HBaseTableUtil, which is I think why this bit was done differently.